app/services/stats/base_monthly_performance_report.rb

```diff
@@ -172,20 +172,27 @@ def create_background_jobs
       @status = Stats::BaseReport::WAITING
     end 
 
-    def merge_stats(data_collector)
-      merged_stats = {}
+    def init_merged_stats
+      merged_result = {}
       (array_of_month_numbers + [:total]).each do |row|
-        merged_stats[row] = {}
+        merged_result[row] = {}
       end
+      merged_result
+    end 
+
+    def merge_stats(data_collector)
+      merged_stats = init_merged_stats()
       data_collector.each do |data|
         data_object = JSON.parse(data, symbolize_names: true)
         data_object.each do |month, stats|
           month_key = month.to_s.to_i > 0 ? month.to_s.to_i : month
           stats.each do |stat_item, value|
-            if !merged_stats[month_key].key?(stat_item)
-              merged_stats[month_key][stat_item] = 0
+            if !merged_stats[month_key].nil?
```

Comment:
Just a style thing, you can probably do 

```suggestion
            if merged_stats[month_key]
```
if you're expecting an object and not specifically trying to differentiate between `false` and `nil`, or 
```suggestion
            if merged_stats[month_key].present?
```
is just a slightly nicer expression than "not is it nil?"

Reply:
@Polsonby ,   the change I made is necessary,  that line is not a change/replacement to the line 185, I added a new line to check whether the month is within the months the report required, then next step is to check whether the metric (such as non_triggered_open_in_time) is in that month or not which is the line 185 trying to do .

Reply:
If not adding this extra check,   an error will be thrown out saying Nil:class doesn't have key method which is the error happened when we demoed to Sarah

Reply:
thank you for your suggestion, actually remind me I should use merged_stats.key?(month_key)